### PR TITLE
Add paygo 5.0 x86_64 BV pipeline 

### PR DESCRIPTION
**Pull request by maximenoel8**
### What does this PR ? 

Add paygo 5.0 Bv pipeline

Depends on: https://github.com/uyuni-project/sumaform/pull/1754

### CODE REVIEW

File: jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy [hunk 1]
```diff
@@ -214,9 +213,9 @@ def run(params) {
...
+                    sh "echo \"export TF_VAR_CUCUMBER_GITREPO=${params.cucumber_gitrepo}; export TF_VAR_CUCUMBER_BRANCH=${params.cucumber_ref}; export TF_VAR_MIRROR=${env.mirror_hostname_aws_private}; export TERRAFORM=${params.terraform_bin}; export TERRAFORM_PLUGINS=${params.terraform_bin_plugins}; export TF_VAR_SERVER_AMI=${server_ami}; export TF_VAR_PROXY_AMI=${proxy_ami}; ./terracumber-cli ${common_params} --logfile ${resultdirbuild}/sumaform-aws.log --init --taint '.*(domain|main_disk).*' --runstep provision --custom-repositories ${WORKSPACE}/custom_repositories.json --sumaform-backend aws\""
+                    retry(count: 1) {
```
Comment (srbarrios) [inline jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy:217]: Having a retry of 1, means re-trying once?
Or we canjust get rid of this function?

File: jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy [hunk 2]
```diff
@@ -231,9 +230,9 @@ def run(params) {
                 }
             }
 
-            stage('Sanity check') {
-                sh "./terracumber-cli ${common_params} --logfile ${resultdirbuild}/testsuite.log --runstep cucumber --cucumber-cmd 'cd /root/spacewalk/testsuite; ${env.exports} rake cucumber:build_validation_sanity_check'"
-            }
+//            stage('Sanity check') {
```
Comment (srbarrios) [inline jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy:233]: Do we miss to uncomment this?

File: jenkins_pipelines/environments/manager-5.0-qe-build-validation-AWS [hunk 3]
```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env groovy
+
+node('sumaform-cucumber') {
+    def minionList = 'sles12sp5_minion, sles12sp5_sshminion, sles12sp5_paygo_minion, ' +
+            'sles15sp4_minion, sles15sp4_sshminion, ' +
+            'sles15sp5_minion, sles15sp5_paygo_minion, ' +
```
Comment (srbarrios) [inline jenkins_pipelines/environments/manager-5.0-qe-build-validation-AWS:6]: Do we also want ssh minions for sp5 and sp6?

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 4]
```diff
@@ -0,0 +1,498 @@
+
+// Mandatory variables for terracumber
+variable "URL_PREFIX" {
+  type = string
+  default = "https://ci.suse.de/view/Manager/view/Manager-43/job/SUSEManager-432-AWS"
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:5]: 50

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 5]
```diff
@@ -0,0 +1,498 @@
...
+}
+
+variable "CUCUMBER_RESULTS" {
+  type = string
+  default = "/root/spacewalk/testsuite"
+}
+
+variable "MAIL_SUBJECT" {
+  type = string
+  default = "Results Manager4.3-WS-MU $status: $tests scenarios ($failures failed, $errors errors, $skipped skipped, $passed passed)"
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:31]: 5.0

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 6]
```diff
@@ -0,0 +1,498 @@
...
+}
+
+variable "MAIL_TEMPLATE" {
+  type = string
+  default = "../mail_templates/mail-template-jenkins.txt"
+}
+
+variable "MAIL_SUBJECT_ENV_FAIL" {
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:39]: 5.0

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 7]
```diff
@@ -0,0 +1,498 @@
...
+  disable_firewall               = false
+  allow_postgres_connections     = false
+  skip_changelog_import          = false
+  create_first_user              = false
+  mgr_sync_autologin             = false
+  create_sample_channel          = false
+  create_sample_activation_key   = false
+  create_sample_bootstrap_script = false
+  publish_private_ssl_key        = false
+  use_os_released_updates        = true
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:205]: I would like to clarify this point, as we had a discussion some days ago where we agreed on the contrary.
Once we clarify, let's add an additional line into our confluence section for agreements about BV, if we need something different for AWS.

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 8]
```diff
@@ -0,0 +1,498 @@
...
+  sles_registration_code = var.SLES_REGISTRATION_CODE
+  use_os_released_updates = false
+  ssh_key_path            = "./salt/controller/id_rsa.pub"
+  provider_settings = {
+    instance_type = "t3a.medium"
+  }
+}
+
+
+module "sles15sp5_sshminion" {
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:402]: We define the modules for ssh minions of sp5 and sp6 here, but we missed above.

File: terracumber_config/tf_files/aws_mirror.tf [hunk 9]
```diff
@@ -1,7 +1,7 @@
 // Mandatory variables for terracumber
 variable "URL_PREFIX" {
   type = string
-  default = "https://ci.suse.de/view/Manager/view/Manager-4.3/job/SUSEManager-4.3-AWS"
+  default = "https://ci.suse.de/view/Manager/view/Manager-5.0/job/SUSEManager-5.0-AWS"
```
Comment (srbarrios) [inline terracumber_config/tf_files/aws_mirror.tf:4]: Actually, we can use this for both versions right? Maybe what we need is an independent pipeline.

File: terracumber_config/tf_files/local_mirror.tf [hunk 10]
```diff
@@ -86,17 +86,18 @@ terraform {
...
 
+
 provider "libvirt" {
-  uri = "qemu+tcp://grog.mgr.prv.suse.net/system"
+  uri = "qemu+tcp://suma-05.mgr.suse.de/system"
 }
 
 locals {
-  pool = "ssd"
+  pool = "mnoel_disks"
```
Comment (srbarrios) [inline terracumber_config/tf_files/local_mirror.tf:100]: I wouldn't use a personal disk for this terraform file, unless it's clear on its filename.

**Review by srbarrios — DISMISSED** [review]

File: jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy (hunk 2 above)
Comment (maximenoel8) [reply jenkins_pipelines/environments/common/pipeline-build-validation-aws.groovy:233]: Sanity check is creating some chaos for now. The first objective of this pipeline is to deploy the infra. The configuration and testing part is not yet compatible. I will remove this sanity step for now. There is probably something to investigate there. The last time I used it, it took several hours to fail.

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf (hunk 7 above)
Comment (maximenoel8) [reply terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:205]: More or less a copy paste of what we have for all the other BVs. Let's open a separate PR if we want to change this.

File: terracumber_config/tf_files/aws_mirror.tf (hunk 9 above)
Comment (maximenoel8) [reply terracumber_config/tf_files/aws_mirror.tf:4]: Done

File: terracumber_config/tf_files/local_mirror.tf (hunk 10 above)
Comment (maximenoel8) [reply terracumber_config/tf_files/local_mirror.tf:100]: done

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf (hunk 7 above)
Comment (srbarrios) [reply terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:205]: A week ago we reach that agreement: https://github.com/uyuni-project/uyuni/pull/9540/files#r1877809363

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 11]
```diff
@@ -0,0 +1,501 @@
...
+    instance_type = "c6i.large"
+  }
+
+  //proxy_additional_repos
+}
+
+module "sles12sp5_paygo_minion" {
+  source             = "./modules/minion"
+  base_configuration = module.base.configuration
+  product_version    = "paygo"
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:241]: That's a valid product_version?

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 12]
```diff
@@ -0,0 +1,501 @@
...
+  auto_connect_to_master  = false
+  use_os_released_updates = false
+  ssh_key_path            = "./salt/controller/id_rsa.pub"
+  additional_packages = [ "python-instance-billing-flavor-check" ]
+}
+
+module "sles15sp5_paygo_minion" {
+  source             = "./modules/minion"
+  base_configuration = module.base.configuration
+  product_version    = "paygo"
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:257]: That's a valid product_version?

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 13]
```diff
@@ -0,0 +1,501 @@
...
+    instance_type = "t3a.medium"
+  }
+  additional_packages = [ "chrony" ]
+
+}
+
+module "sles15sp4_minion" {
+  source             = "./modules/minion"
+  base_configuration = module.base.configuration
+  product_version    = "4.3-released"
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:326]: Shouldn't that be 5.0-released?

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf [hunk 14]
```diff
@@ -0,0 +1,501 @@
...
+  provider_settings = {
+    instance_type = "t3a.medium"
+  }
+
+}
+
+module "sles15sp5_minion" {
+  source             = "./modules/minion"
+  base_configuration = module.base.configuration
+  product_version    = "4.3-released"
```
Comment (srbarrios) [inline terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:343]: Shouldn't that be 5.0-released?

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf (hunk 12 above)
Comment (srbarrios) [reply terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:257]: There are few more like those.

Comment (maximenoel8) [reply terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:257]: yes, it's to trigger specific tests in the testsuite

File: terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf (hunk 13 above)
Comment (maximenoel8) [reply terracumber_config/tf_files/SUSEManager-5.0-build-validation-paygo-AWS.tf:326]: indeed

**Pull request merged**